### PR TITLE
Adding wait method to LRO pollers

### PR DESCRIPTION
**Pull request by catalinaperalta**
Adding the Wait method on LRO pollers, add Wait to LRO tests and remove extra nil check in response() method.

### CODE REVIEW

File: src/generator/pollers.ts [hunk 1]
```diff
@@ -24,17 +24,20 @@ export async function generatePollers(session: Session<CodeModel>): Promise<stri
...
   imports.add('time');
+  imports.add('strconv');
   text += imports.text();
 
   const pollers = <Array<PollerInfo>>session.model.language.go!.pollerTypes;
   pollers.sort((a: PollerInfo, b: PollerInfo) => { return sortAscending(a.name, b.name) });
   for (const poller of values(pollers)) {
     const pollerInterface = pascalCase(poller.name);
-    let responseType = '';
+	let responseType = '';
```
Comment (jhendrixMSFT) [inline src/generator/pollers.ts:34]: Nit, spacing (next line too).

File: src/generator/pollers.ts [hunk 2]
```diff
@@ -74,15 +77,34 @@ func (p *${poller.name}) Poll(ctx context.Context) (*${responseType}, error) {
...
-	return nil, nil
+	for {
+		resp, err := p.Poll(context.Background())
+		if err != nil {
+			return nil, err
+		}
+		if p.Done() {
+			return resp, err
+		}
+		if ra := resp${rawResponse}.Header.Get(azcore.HeaderRetryAfter); len(ra) > 0 {
```
Comment (jhendrixMSFT) [inline src/generator/pollers.ts:92]: You can use `resp.RetryAfter()` which has all this code in it already.

Comment (catalinaperalta) [reply src/generator/pollers.ts:92]: SO I had the same thought and tried to do that until I realized that the response that we receive here is the response envelope for the operation.

File: src/generator/pollers.ts (hunk 1 above)
Comment (catalinaperalta) [reply src/generator/pollers.ts]: Done

File: src/generator/pollers.ts (hunk 2 above)
Comment (jhendrixMSFT) [reply src/generator/pollers.ts:92]: Per our conversation can use `p.response()`.

Comment (catalinaperalta) [reply src/generator/pollers.ts]: Done
